src/cpu/x64/matmul/brgemm_matmul_reorders.cpp

```diff
@@ -79,8 +79,10 @@ status_t calculate_plain_transpose_blocks(dim_t &batch, dim_t &M, dim_t &K,
     }
 
     memory_desc_t src_md_reduced, dst_md_reduced;
-    memory_desc_reshape(src_md_reduced, src_md, non_unit_dim, non_unit_dims);
-    memory_desc_reshape(dst_md_reduced, dst_md, non_unit_dim, non_unit_dims);
+    CHECK(memory_desc_reshape(
+            src_md_reduced, src_md, non_unit_dim, non_unit_dims));
+    CHECK(memory_desc_reshape(
```

Comment:
Looks like we have some duplicate work with #3095/#3080, except I ended up using VDISPATCH_REORDER_IC to be aligned with the error behavior in the rest of the function. Would you mind just pulling in the commit from #3095 to this PR reduce validation?

Reply:
Sorry about it, I wanted to remove numerical issue to validate my changes and just left it there. I've cherry-picked your backport commit. Feel free to rebase a PR to main and commit on top. Thanks for letting me know.